foglamp/FogLAMP
# FOGL-2386 - Python South Async plugins having FogLAMP req as >= 1.5 should use south_c as process_name and should disable the schedule upon upgrade

Steps to test:
1. Start foglamp and add a Python async service (e.g. sinusoid async_backup_south_c branch) in disabled mode.
2. In DB, change process_name to "south" in schedules table for the service. You may be required to create an entry in scheduled_processes table.
3. Enable service. Note syslog. After a while, service will start running with south_c server.
4. Confirm process_name in schedules table for service. It should show "south_c".

Tested with sinusoid plugin:
```
Feb  7 16:42:10 asinha-ThinkPad-E450 FogLAMP[20906] WARNING: server: foglamp.services.south.server: Plugin play should be loaded by C version of South server
Feb  7 16:42:11 asinha-ThinkPad-E450 FogLAMP[20794] INFO: utils: foglamp.services.common.utils: Ping received for Service play id 4bfe1c6d-5b69-4d3d-a01f-1fe72d76e478 at url http://0.0.0.0:44757/foglamp/service/ping
Feb  7 16:42:11 asinha-ThinkPad-E450 FogLAMP[20794] INFO: utils: foglamp.services.common.utils: Shutting down the Southbound service play ...
Feb  7 16:42:11 asinha-ThinkPad-E450 FogLAMP[20794] INFO: utils: foglamp.services.common.utils: Service play, id 4bfe1c6d-5b69-4d3d-a01f-1fe72d76e478 at url http://0.0.0.0:44757/foglamp/service/shutdown successfully shutdown
Feb  7 16:42:11 asinha-ThinkPad-E450 FogLAMP[20794] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Disabled Schedule 'play/ca870c13-735d-4067-9a00-466e4e161da8' process 'south'
Feb  7 16:42:11 asinha-ThinkPad-E450 FogLAMP[20906] INFO: modify_process: foglamp.services.south.modify_process: Disabled Python South service [play] to change process_name to south_c
Feb  7 16:42:11 asinha-ThinkPad-E450 FogLAMP[20794] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Service play records successfully removed
Feb  7 16:42:11 asinha-ThinkPad-E450 FogLAMP[20906] INFO: modify_process: foglamp.services.south.modify_process: Modified process_name from "south" to "south_c" for Python South service [play]: {"schedule": {"name": "play", "type": "STARTUP", "day": null, "id": "ca870c13-735d-4067-9a00-466e4e161da8", "exclusive": true, "time": 0, "repeat": 0, "enabled": false, "processName": "south_c"}}
Feb  7 16:42:11 asinha-ThinkPad-E450 FogLAMP[20906] INFO: playback: foglamp.plugins.south.playback.playback: playback plugin shut down.
Feb  7 16:42:12 asinha-ThinkPad-E450 FogLAMP[20794] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Scheduled task for schedule 'play' to start at 2019-02-07 16:42:12.803130
Feb  7 16:42:12 asinha-ThinkPad-E450 FogLAMP[20794] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Enabled Schedule 'play/ca870c13-735d-4067-9a00-466e4e161da8' process 'south_c'
Feb  7 16:42:12 asinha-ThinkPad-E450 FogLAMP[20794] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process started: Schedule 'play' process 'south_c' task 60c8c6c4-233d-4795-bb9d-290df9aef18c pid 20913, 1 running tasks#012['services/south_c', '--port=42423', '--address=127.0.0.1', '--name=play']
Feb  7 16:42:12 asinha-ThinkPad-E450 FogLAMP[20906] INFO: modify_process: foglamp.services.south.modify_process: Enabled Python South service [play] after changing process_name to south_c
Feb  7 16:42:13 asinha-ThinkPad-E450 play[20915]: INFO: PluginInterfaceInit:81: shimLayerPath=/home/asinha/dev/FogLAMP/python/foglamp/plugins/common/shim, foglampPythonDir=/home/asinha/dev/FogLAMP/python
Feb  7 16:42:14 asinha-ThinkPad-E450 FogLAMP[20794] INFO: service_registry: foglamp.services.core.service_registry.service_registry: Registered service instance id=4bfe1c6d-5b69-4d3d-a01f-1fe72d76e478: <play, type=Southbound, protocol=http, address=localhost, service port=None, management port=45241, status=1>
Feb  7 16:42:14 asinha-ThinkPad-E450 FogLAMP[20915] INFO: playback: foglamp.plugins.south.playback.playback: plugin_start called
Feb  7 16:42:14 asinha-ThinkPad-E450 FogLAMP[20915] WARNING: playback: foglamp.plugins.south.playback.playback: playback - EOF reached:
```
